### ISSUE TITLE
Add examples for making your own dataset

### ISSUE THREAD

**Issue by derek-saal** [rusty1s/pytorch_geometric#27]

**Comment by derek-saal** [comment]
Actually its more straightforward than I thought... I will repoen if I have any specific questions.

Im reading the [Guide](https://rusty1s.github.io/pytorch_geometric/build/html/notes/create_dataset.html#creating-in-memory-datasets) in split screen with the [examples](https://github.com/rusty1s/pytorch_geometric/tree/master/torch_geometric/datasets).

**Comment by derek-saal** [comment]
Im a bit confused on what edge_attr is. Some definitions or requirements would be really helpful @rusty1s .

**Comment by rusty1s** [comment]
`edge_attr` contain the edge features with shape `[E, D]` (see https://rusty1s.github.io/pytorch_geometric/build/html/notes/introduction.html).

**Comment by derek-saal** [comment]
> `edge_attr` contain the edge features with shape `[E, D]` (see https://rusty1s.github.io/pytorch_geometric/build/html/notes/introduction.html).

Im not sure how I missed that... Thanks. Ill give it a shot and let you know if I get stuck.

**Comment by derek-saal** [comment]
@rusty1s , do all nodes need to be for training or testing? What would be the process to add nodes that are supporting?

**Comment by rusty1s** [comment]
You can mask out nodes that are not relevant to the loss function (see https://github.com/rusty1s/pytorch_geometric/blob/master/examples/cora.py)

**Comment by fighting41love** [comment]
Thanks for sharing the pytorch_geometirc.
I have a question for making my own dataset. 
Take the three nodes figure ([link](
https://rusty1s.github.io/pytorch_geometric/build/html/_images/graph.svg) )as an example.
The corresponding code is as follows.
`edge_index = torch.tensor([[0, 1, 1, 2],
                           [1, 0, 2, 1]], dtype=torch.long)`
I searched COO format in google, and got nothing.
Could you pls help to figure out what is the COO format, and why the figure is defined as a (2*4) matrix?
Thanks!

**Comment by fighting41love** [comment]
For a graph with 4 nodes (not the open triangle in the above figure), for instance [link](https://dianafea.com/manuals/d96/Theory/img270.png). 
The edge_index is defined as:
`edge_index = torch.tensor([[1, 2, 2, 3,3,4,4,1], [2, 1, 3, 2,4,3,1,4]], dtype=torch.long)
`
Is it right?
If it is right, what is the motivation to design edge_index in a (2*4) shape?
Thanks!

**Comment by fighting41love** [comment]
> Hi,
> 
> we represent adjacency matrices by `edge_index` and `edge_attr` similar to the sparse tensor format of PyTorch (https://pytorch.org/docs/stable/sparse.html), except that we do not wrap it in a `torch.sparse_coo_tensor` (we can differentiate w.r.t. `edge_attr`, something which is currently not supported for sparse tensors in PyTorch). You can read more about the coo format [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html) or [here](https://www.scipy-lectures.org/advanced/scipy_sparse/coo_matrix.html).
> 
> The reason to design `edge_index` as a [2, E] tensor instead of [E, 2] is the contiguous memory representation of the `row` and `col` data.
> 
> Your example looks correct, however, indices should start at zero:
> 
> ```python
> edge_index = edge_index - 1
> ```

Thanks for the quick response! Problem solved. Thanks!
